UnitTests/UnitTests.UWP/UI/Triggers/Test_ControlSizeTrigger.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Toolkit.Uwp;
+using Microsoft.Toolkit.Uwp.UI.Triggers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Threading.Tasks;
+using Windows.UI.Xaml.Controls;
+
+namespace UnitTests.UWP.UI.Triggers
+{
+    [TestClass]
+    [TestCategory("Test_ControlSizeTrigger")]
+    public class Test_ControlSizeTrigger : VisualUITestBase
+    {
+        [DataTestMethod]
+        [DataRow(450, 450, true)]
+        [DataRow(400, 400, true)]
+        [DataRow(500, 500, false)]
+        [DataRow(399, 400, false)]
+        [DataRow(400, 399, false)]
+        public async Task ControlSizeTriggerTest(double width, double height, bool expectedResult)
+        {
+            await App.DispatcherQueue.EnqueueAsync(() =>
+            {
+                Grid grid = CreateGrid(width, height);
+                var trigger = new ControlSizeTrigger();
+
+                trigger.TargetElement = grid;
+                trigger.MaxHeight = 500;
+                trigger.MinHeight = 400;
+                trigger.MaxWidth = 500;
+                trigger.MinWidth = 400;
+
+                Assert.AreEqual(expectedResult, trigger.IsActive);
```

Comment:
Yeah failing here:
```
Failed ControlSizeTriggerTest [578 ms]
  Failed ControlSizeTriggerTest (450,450,True) [575 ms]
  Error Message:
   Assert.AreEqual failed. Expected:<True>. Actual:<False>. 
  Stack Trace:
     at UnitTests.UWP.UI.Triggers.Test_ControlSizeTrigger.<>c__DisplayClass0_0.<ControlSizeTriggerTest>b__0() + 0x11c
   at System.Action.Invoke() + 0x1d
   at Microsoft.Toolkit.Uwp.DispatcherQueueExtensions.<>c__DisplayClass1_0.<EnqueueAsync>b__1() + 0x20
--- End of stack trace from previous location where exception was thrown ---
```

Even though you're using `VisualUITestBase` you never call `SetTestContentAsync`, so it's not actually part of the Visual Tree. I imagine some 'magic' may happen with the app running locally but this may be what's causing failure for the grid's actual size in the CI?

I'd add that to put the Grid in app vs. relying on manually calling the update layout methods, see if that makes a difference?

Reply:
Ah I didn't know about that method. Started using and tests are passing locally. Fingers crossed the CI builds pass too.